lldb/source/Plugins/Language/Swift/SwiftLanguage.cpp

```diff
@@ -1761,95 +1762,104 @@ bool SwiftLanguage::GetFunctionDisplayName(
       s << display_name;
       return true;
     }
-    const char *cstr = display_name.data();
-    const char *open_paren = strchr(cstr, '(');
-    const char *close_paren = nullptr;
-    const char *generic = strchr(cstr, '<');
-    // If before the arguments list begins there is a template sign
-    // then scan to the end of the generic args before you try to find
-    // the arguments list.
-    if (generic && open_paren && generic < open_paren) {
-      int generic_depth = 1;
-      ++generic;
-      for (; *generic && generic_depth > 0; generic++) {
-        if (*generic == '<')
-          generic_depth++;
-        if (*generic == '>')
-          generic_depth--;
-      }
-      if (*generic)
-        open_paren = strchr(generic, '(');
-      else
-        open_paren = nullptr;
-    }
-    if (open_paren) {
-      close_paren = strchr(open_paren, ')');
+    return GetFunctionDisplayArgs(args, s, display_name, exe_scope);
+  }
+  }
+  return false;
+}
+
+bool SwiftLanguage::GetFunctionDisplayArgs(
+    VariableList &args, lldb_private::Stream &s, std::string display_name,
+    lldb_private::ExecutionContextScope *exe_scope) {
+  const char *cstr = display_name.data();
+  const char *open_paren = strchr(cstr, '(');
```

Comment:
llvm::StringRef::consume() and find() would be safer alternatives to the C stdlib functions.

Reply:
The bulk of the changes to this file were ported to https://github.com/swiftlang/llvm-project/pull/10766. Once it's merged, these changes should be removed from the diff.